.github/workflows/automated-testing.yml

```
@@ -0,0 +1,351 @@
+        run: |
+          cd frontend
+          npm run dev &
+          sleep 10
+
+      - name: Run E2E tests on ${{ matrix.browser }}
+        run: |
+          cd e2e
+          npx playwright test --project=${{ matrix.browser }} --reporter=html
+
+      - name: Upload E2E test results
+        uses: actions/upload-artifact@v3
```

actionlint (1.7.7)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)